cpp/src/io/json/reader_impl.cu

```diff
@@ -50,6 +50,15 @@ namespace json {
 using namespace cudf::io;
 
 namespace {
+/**
+ * @brief Helper class to support inline-overloading for all of a variant's alternative types
+ */
+template <class... Ts>
+struct VisitorOverload : Ts... {
+  using Ts::operator()...;
+};
+template <class... Ts>
+VisitorOverload(Ts...) -> VisitorOverload<Ts...>;
```

Comment:
This is copy-pasted from CSV, not sure what is the right common location for this.

Reply:
Being a more generic helper class, maybe somewhere in `cpp/include/cudf/utilities/...`?